SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/TypeStructure.cpp

```diff
@@ -491,25 +527,37 @@ FSubobjectMap GetAllSubobjects(TSharedPtr<FUnrealType> TypeInfo)
 	TSet<UObject*> SeenComponents;
 	uint32 CurrentOffset = 1;
 
+	auto AddSubobject = [&CurrentOffset, &SeenComponents, &Subobjects](TSharedPtr<FUnrealType> PropertyTypeInfo) {
+		UObject* Value = PropertyTypeInfo->Object;
+
+		if (Value != nullptr && IsSupportedClass(Value->GetClass()))
+		{
+			if (!SeenComponents.Contains(Value))
+			{
+				SeenComponents.Add(Value);
+				Subobjects.Add(CurrentOffset, PropertyTypeInfo);
+			}
+
+			CurrentOffset++;
```

Comment:
Should we increment the offset here? This code is moved from below, but still seems like something that we shouldn't have to do.

Reply:
It should be safe to remove the offset stuff altogether - they're not used anymore. You can just return an array of `TSharedPtr<FUnrealType>` instead of `FSubobjectMap` - the keys in the map are ignored.

Reply:
I'm not sure why we're incrementing the offset when we don't add to the `SeenComponents`. I can see this is legacy code though - @improbable-valentyn do you know why?